### PR TITLE
Add suspended as option to AdminService.CreateUser()

### PR DESCRIPTION
**Pull request by ttomsu**
`suspended` value can be found here: https://docs.github.com/en/enterprise-server@3.11/rest/enterprise-admin/users?apiVersion=2022-11-28#create-a-user

### CODE REVIEW

File: github/admin_users.go [hunk 1]
```diff
@@ -13,21 +13,23 @@ import (
...
+	Suspended *bool   `json:"suspended,omitempty"`
 }
 
 // CreateUser creates a new user in GitHub Enterprise.
 //
 // GitHub API docs: https://docs.github.com/enterprise-server@3.11/rest/enterprise-admin/users#create-a-user
 //
 //meta:operation POST /admin/users
-func (s *AdminService) CreateUser(ctx context.Context, login, email string) (*User, *Response, error) {
+func (s *AdminService) CreateUser(ctx context.Context, login, email string, suspended bool) (*User, *Response, error) {
```
Comment (gmlewis) [inline github/admin_users.go:26]: From the official docs: https://docs.github.com/en/enterprise-server@3.11/rest/enterprise-admin/users?apiVersion=2022-11-28#create-a-user
I'm noticing that both the `email` and `suspended` fields are optional.

Since we are breaking the API with this PR, I'm wondering if it might make more sense to create a new `CreateUserOptions` struct containing these optional fields?  In addition, `login` is required, so maybe change line 16 above to `Login string `json:"login"`.

In other words, we would still use the private `createUserRequest` but we would copy fields in from `opts` where the new signature would be:

```go
unc (s *AdminService) CreateUser(ctx context.Context, login string, opts *CreateUserOptions) (*User, *Response, error) {
```

If we do this now, then we might have a bit more flexibility in the future to add new fields to the `CreateUserOptions` struct without breaking the API again.

Thoughts?

Comment (ttomsu) [reply github/admin_users.go:26]: Hi Glenn - I like the idea to break it only once if we have to break it at all.

Taking a look at other examples around the codebase, it looks like one pattern is to pass URL path segments individually, then the whole object that would end up in the request body. Examples

* [TeamsService.CreateTeam(ctx context.Context, org string, team NewTeam)](https://github.com/google/go-github/blob/e9f52699f5e55a4791567bce668f56446e6343ef/github/teams.go#L186)
* [PullRequestService.Create(ctx context.Context, owner string, repo string, pull *NewPullRequest)](https://github.com/google/go-github/blob/e9f52699f5e55a4791567bce668f56446e6343ef/github/pulls.go#L268)

Following this pattern, it seems a more consistent option would be to export the `createUserRequest` object and pass that as the second param (after `ctx`). 

WDYT?

Comment (gmlewis) [reply github/admin_users.go:26]: Ah, yes, I see what you are saying.  Path query parameters should be listed as separate arguments, and then you can have another parameter:

```go
func (s *AdminService) CreateUser(ctx context.Context, req CreateUserRequest) (*User, *Response, error) {
```

where:

```go
// CreateUserRequest represents the fields sent to the `CreateUser` endpoint.
// Note that `Login` is a required field.
type CreateUserRequest struct {
	Login     string `json:"login"`
	Email     *string `json:"email,omitempty"`
	Suspended *bool   `json:"suspended,omitempty"`
}
```

**Comment by gmlewis** [comment]
In the future, please don't use force-push in this repo, as we always squash-and-merge at the end.  See CONTRIBUTING.md for more details.  Thanks.

File: github/admin_users.go [hunk 2]
```diff
@@ -10,26 +10,25 @@ import (
...
 
-// createUserRequest is a subset of User and is used internally
-// by CreateUser to pass only the known fields for the endpoint.
-type createUserRequest struct {
-	Login *string `json:"login,omitempty"`
-	Email *string `json:"email,omitempty"`
+// CreateUserRequest represents the fields sent to the `CreateUser` endpoint.
+// Note that `Login` is a required field.
+type CreateUserRequest struct {
+	Login     *string `json:"login"`
```
Comment (gmlewis) [inline github/admin_users.go:16]:
```suggestion
	Login     string `json:"login"`
```

File: github/admin_users.go [hunk 3]
```diff
@@ -10,26 +10,25 @@ import (
...
+	Email     *string `json:"email,omitempty"`
+	Suspended *bool   `json:"suspended,omitempty"`
+}
+
+type CreateUserOptions struct {
```
Comment (gmlewis) [inline github/admin_users.go:21]: Please delete.

File: github/admin_users.go [hunk 4]
```diff
@@ -10,26 +10,25 @@ import (
...
 }
 
 // CreateUser creates a new user in GitHub Enterprise.
 //
 // GitHub API docs: https://docs.github.com/enterprise-server@3.11/rest/enterprise-admin/users#create-a-user
 //
 //meta:operation POST /admin/users
-func (s *AdminService) CreateUser(ctx context.Context, login, email string) (*User, *Response, error) {
+func (s *AdminService) CreateUser(ctx context.Context, userReq *CreateUserRequest) (*User, *Response, error) {
```
Comment (gmlewis) [inline github/admin_users.go:29]: Since this is required information, let's please pass by value, not by reference:

```suggestion
func (s *AdminService) CreateUser(ctx context.Context, userReq CreateUserRequest) (*User, *Response, error) {
```

**Review by gmlewis — CHANGES_REQUESTED** [review]

File: github/admin_users.go (hunk 2 above)
Comment (ttomsu) [reply github/admin_users.go]: Done.

File: github/admin_users.go (hunk 3 above)
Comment (ttomsu) [reply github/admin_users.go]: Done.

File: github/admin_users.go (hunk 4 above)
Comment (ttomsu) [reply github/admin_users.go]: Done.

**Comment by ttomsu** [comment]
Apologies for the force commit, my local branch got out of whack and I didn't realize it.

**Comment by ttomsu** [comment]
@gmlewis Anyone you can ping to get a second set of eyes on?

**Comment by gmlewis** [comment]
All contributors to this repo are volunteers and I typically try not to ping individuals unless there is an urgent need.

**Comment by fchimpan** [comment]
Hi, @ttomsu !
Thanks your implementation. LGTM! Great job!

> If we do this now, then we might have a bit more flexibility in the future to add new fields to the CreateUserOptions struct without breaking the API again.

I agree it. It's a good design.

**Comment by gmlewis** [comment]
Thank you, @fchimpan !
Merging.
